### DISCUSSION TITLE
About single instance performance

### DISCUSSION THREAD

**Discussion by 1998729** [jertel/elastalert2#1072] [Q&A] [answered]
Have you tested the number of rule files carried by an elastalert?

For example, if you query a rule for 3 minutes in 1 minute, If the performance of the es cluster is fine, how many rule files can elastalert2 carry?

**Comment by 1998729** [comment]
Our frequency type rule files will increase in the future, and I am not sure how many files a single instance can carry.

All query conditions are as follows:

index:xxxxxx-%Y.%m.%d
use_strftime_index: true

run_every:
   minutes: 1

buffer_time:
   minutes: 3

num_events: 1
timeframe:
   minutes: 1

**Comment by jertel** [comment, marked as answer]
ElastAlert 2 is primarily limited by the Elasticsearch cluster performance. Running several thousand rules is not uncommon. But it requires that each rule is carefully tuned to only query and return what is absolutely necessary, and that the Elasticsearch cluster is properly maintained, has sufficient RAM and CPU on each cluster node, index fields are tuned for performance, and index life-cycles are configured to avoid placing unnecessary load on the cluster. 

See #187 and #948 for similar questions.
